defi/src/updateSearch.ts

```diff
@@ -249,10 +268,111 @@ const getProtocolSubSections = ({
   }));
 };
 
+async function getAllCurrentSearchResults() {
+  const res: { total: number; results: Array<SearchResult> } = await fetch(
+    "https://search.defillama.com/indexes/pages/documents",
+    {
+      headers: {
+        Authorization: `Bearer ${process.env.SEARCH_MASTER_KEY}`,
+      },
+    }
+  ).then((res) => res.json());
+  const finalRes: { total: number; results: Array<SearchResult> } = await fetch(
+    `https://search.defillama.com/indexes/pages/documents?limit=${res.total}`,
+    {
+      headers: {
+        Authorization: `Bearer ${process.env.SEARCH_MASTER_KEY}`,
+      },
+    }
+  ).then((res) => res.json());
+  return finalRes.results;
+}
+
+function getResultsToDelete(currentResults: Array<SearchResult>, newResults: Array<SearchResult>) {
+  const newResultsSet = new Set(newResults.map((r) => r.route));
+
+  return currentResults
+    .filter((item) => {
+      if (newResultsSet.has(item.route)) {
+        return false;
+      }
+
+      if (item.type === "Protocol") {
+        // can delete as the protocol is no longer in the list (maybe delisted or renamed)
+        if (!protocolNamesSet.has(item.name)) {
+          return true;
+        }
+
+        // it is okay to delete if it is a sub section page, maybe we no longer track some metric for this protocol
+        if (item.subName) {
+          return true;
+        }
+      }
+
+      if (item.type === "Chain") {
+        const newChainName = getChainDisplayName(item.name, true);
+        // delete if the chain name has changed
+        if (newChainName !== item.name) {
+          return true;
+        }
+
+        // it is okay to delete if it is a sub section page, maybe we no longer track some metric for this chain
+        if (item.subName) {
+          return true;
+        }
+      }
+
+      // only delete if no protocols are in this category
+      if (item.type === "Category" && !categoriesSet.has(item.name) && !(CategoryTagMap as any)[item.name]) {
+        return true;
+      }
+
+      // only delete if no protocols are in this tag
+      if (item.type === "Tag" && !tagsSet.has(item.name)) {
+        return true;
+      }
+
+      if (item.type === "Stablecoin") {
+        return true;
+      }
+
+      // it is okay to delete as this relies on frontend pages, can't really be sure if it is still valid
+      if (item.type === "Metric") {
+        return true;
```

Comment:
isnt this gonna create a similar issue as we had before, where if one query succeeds but the other doesnt it will lead to missing results?